### ISSUE TITLE
Order stream timing?

### ISSUE THREAD

**Issue by fartmanly** [JKorf/Binance.Net#56]
Hello,
I'm noticing that when 2 orders are executed within a short time of each other (less than 5 seconds), the 2nd order is not detected by the order stream subscription.
I've tested this using my phone to buy and sell orders in the binance app. If I try to cancel 2 orders very quickly, only the first cancel is detected. The 2nd cancel does not even fire the order message event.
Same thing happens whether it's a new buy or sell order or whatever.
However, if I wait at least 5 seconds between the 2 orders, the events seem to be properly detected.

Is there an issue firing order events if they are too close together?

Sorry if I'm not explaining this properly.
I'm using Visual Studio 2017 and programming in VB.net.

Thank you,
:)
Dean

**Comment by JKorf** [comment]
Hi, I can't reproduce this, if I cancel orders on the Binance website the update events get in as I cancel them, no matter how fast I do it. How are you handling the update event?

**Comment by fartmanly** [comment]
Here is basically the code I'm using to subscribe to the order stream (I've removed error checking, etc just for this example). The code works and I've set a breakpoint at the first line of the msgReceived_Order routine so I immediately know if the event has fired. However, if events happen too quickly, it is never called. If I wait a few seconds between events, it seems to work ok.

	Private Sub SubscribeMyOrderStream()

		Dim binSocketClient As New BinanceSocketClient
		Dim listenKey As String = binClient.StartUserStream.Data.ListenKey

		Dim oOrderUpdate As New Action(Of BinanceStreamOrderUpdate)(Sub(orderUpdate As BinanceStreamOrderUpdate)
																																	msgReceived_Order(orderUpdate)
																																End Sub)

		Dim result As CallResult(Of BinanceStreamSubscription) = Nothing
		result = binSocketClient.SubscribeToUserStream(listenKey, Nothing, oOrderUpdate)

		If result.Success Then
			MyOrderStreamSubscription = result.Data
		Else
			'process error here
		End If

	End Sub

	Private Sub msgReceived_Order(e As BinanceStreamOrderUpdate)
		'process event here
	End Sub

**Comment by JKorf** [comment]
I've found the issue. An exception occurred because of the Nothing being passed. That caused the client to disconnect from the server, but then automatically reconnect. The reconnecting takes 5 seconds. If you got an event in those 5 seconds it wouldn't come in, but after that it was reconnected so it would work again.

I've pushed a fix which handles Nothing/Null being passed into the function, so it should work now
